docs/node/grpc.md

```diff
@@ -0,0 +1,62 @@
+---
+description: gRPC proxy for Oasis node
+---
+
+# gRPC proxy for your Oasis node
+
+The Oasis node uses a [gRPC interface] which enables external applications such
+as the Oasis CLI, cross-layer and Oasis Wasm dApps, native wallets, indexers,
+system monitors, health checks and similar to access various consensus and
+ParaTime services.
+
+:::info
+
+The RPC interface is ONLY exposed via an AF_LOCAL socket
+called `internal.sock` located in the node's data directory.
+**This interface should NEVER be directly exposed over the network as it has no
+authentication and allows full control, including shutdown, of a node.**
+
+:::
+
+This chapter shows you how to **tunnel a Unix socket over the network**, so it
+can safely be used on your client, or set up a **public** gRPC endpoint using
+Envoy or NginX and expose only a **safe subset of the [Oasis RPC services]**.
+
+## Tunnel Unix socket via SSH
+
+SSH supports forwarding a Unix socket over a secure layer. The command below
+will establish a secure shell to the `my-oasis-node.com` server and then tunnel
+the `internal.sock` file inside the data directory to a Unix socket inside your
+home folder:
+
+```shell
+ssh oasis@my-oasis-node.com -L /home/user/oasis-node-internal.sock:/node/data/internal.sock
+```
+
+The `/home/user/oasis-node-internal.sock` can now be used to safely connect to
+the Oasis node **as if it was running locally without filtering any services**.
+For example, using the [Oasis CLI]:
+
+```shell
+oasis network add-local my-oasis-node unix:/home/user/oasis-node-internal.sock
+```
+
+```shell
+oasis network status --network my-oasis-node
+```
+
+:::tip Permanent SSH channel
+
+To make a tunneled Unix socket over SSH permanent, consider using [autossh].
+
+:::
+
+[Oasis CLI]: https://github.com/oasisprotocol/cli/blob/master/docs/network.md#add-local
+[autossh]: https://www.harding.motd.ca/autossh/
+
+## Set up gRPC Proxy with Envoy {#envoy}
+
+## Set up gRPC Proxy with Nginx {#nginx}
```

Comment:
Maybe also [Caddy](https://caddyserver.com/).

Reply:
Let's do this in a separate PR. I also removed Nginx from the chapter.